src/smbclient/_os.py

```diff
@@ -1172,6 +1172,24 @@ def _set_basic_information(
         set_info(transaction, basic_info)
 
 
+class LocalDirEntry:
```

Comment:
I would probably put this in `shutil.py` as it's not SMB related but rather a shim used by the shutil function. I'm trying to keep `_os.py` strictly related to SMB operations. I'm also wondering if it's even needed as we could just utilise the result of [os.scandir](https://docs.python.org/3/library/os.html#os.scandir) which returns a [DirEntry](https://docs.python.org/3/library/os.html#os.DirEntry) object I modeled `SMBDirEntry` from.